src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2232,8 +2241,11 @@ impl Coordinator {
             in_immediate_multi_stmt_txn,
             key,
             typ,
+            dataflow_metainfo,
         }: PeekStageTimestamp,
     ) -> Option<(ExecuteContext, PeekStageFinish)> {
+        self.process_optimizer_notices(ctx.session(), &dataflow_metainfo.optimizer_notices);
```

Comment:
This needs to be moved down next to emit_timestamp_notice.

Reply:
Thanks, moved it there.